Add PngSuite test cases

rlepigre opened the pull request:
See http://www.schaik.com/pngsuite/.

This partially addresses https://github.com/rlepigre/ocaml-imagelib/issues/13.

rlepigre committed:
Add PngSuite-based test cases.

rlepigre committed:
Improve error handing in [imagelib-convert].

mseri commented:
Adding `"conf-libMagickCore" {with-test}` to the dependencies should make `magick` available for ghe testa and ensure it will lso be there for the opam repository ci at the next release

rlepigre committed:
Add missing test dependency on magick.

rlepigre commented:
This catches a few bugs that I'll try to fix before merging. One is a parsing bug which I'm pretty sure was not there in the initial version of the library, so it is a regression.

**Edit:** this was indeed a regression introduced in https://github.com/rlepigre/ocaml-imagelib/commit/b30a071621a89b476baad7af34c7b86482cd8ed9#r164902611.

rlepigre committed:
Fix breaking PNG bug.

It was introduced in b30a071621a89b476baad7af34c7b86482cd8ed9.

rlepigre committed:
Slight internal refactoring in PNG.

rlepigre committed:
More precise error message in PNG printer.

rlepigre committed:
Promote broken tests for now.

The problem has to do with transparency, and seems to be a
printer issue in both cases.

rlepigre committed:
[CI] install imagemagick.

rlepigre committed:
Tweak magick compare output.

rlepigre commented:
> Adding `"conf-libMagickCore" {with-test}` to the dependencies should make `magick` available for ghe testa and ensure it will lso be there for the opam repository ci at the next release

@mseri unfortunately that does not work, because the tests need the `magick` tool available, whereas `conf-libMagickCore` only pulls in the library. I removed the dependency for now, since it does not help.

rlepigre commented on tests/png-suite/tests/transparency.t/run.t:
```diff
@@ -0,0 +1,30 @@
+  $ ../test.sh *.png
+  Testing tbbn0g04.png
+  Corrupted image: cannot write greyscale image with alpha to PNG with bitdepth of 4.
```
This error happens when writing the output PNG, and has to do with the fact that we need to emit a `tRNS` chunk for this particular image. We should fix this, but not in this PR.

rlepigre commented on tests/png-suite/tests/transparency.t/run.t:
```diff
@@ -0,0 +1,30 @@
+  Testing tbgn3p08.png
+  0
+  Testing tbrn2c08.png
+  453
```
I'm not sure why there is a difference here, but my guess is that this again has to do with `tRNS` handling. I suspect the issue is again with the PNG output, but I did not confirm.
